scripts/Muon/GUI/Common/load_run_widget/load_run_presenter.py

```diff
@@ -219,33 +218,39 @@ def handle_load_thread_start(self, filenames, finished_callback):
         self._load_thread = self.create_load_thread()
         self._load_thread.threadWrapperSetUp(self.disable_loading,
                                              finished_callback,
-                                             self._view.warning_popup)
+                                             self.error_callback)
         self._load_thread.loadData(filenames)
         self._load_thread.start()
 
-    def handle_load_thread_finished(self):
+    def error_callback(self, error_message):
+        self.enable_notifier.notify_subscribers()
+        self._view.warning_popup(error_message)
 
+    def handle_load_thread_finished(self):
         self._load_thread.deleteLater()
         self._load_thread = None
 
         self.on_loading_finished()
 
     def on_loading_finished(self):
-        if self.run_list and self.run_list[0] == 'Current':
-            self.run_list = [self._model._loaded_data_store.get_latest_data()['run']][0]
-            self._model.current_run = self.run_list
-
-        run_list = [[run] for run in self.run_list if self._model._loaded_data_store.get_data(run=[run])]
-        self._model._context.current_runs = run_list
+        try:
+            if self.run_list and self.run_list[0] == 'Current':
+                self.run_list = [self._model._loaded_data_store.get_latest_data()['run']][0]
+                self._model.current_run = self.run_list
 
-        if self._load_multiple_runs and self._multiple_file_mode == "Co-Add":
-            run_list_to_add = [run for run in self.run_list if self._model._loaded_data_store.get_data(run=[run])]
-            run_list = [[run for run in self.run_list if self._model._loaded_data_store.get_data(run=[run])]]
-            load_utils.combine_loaded_runs(self._model, run_list_to_add)
+            run_list = [[run] for run in self.run_list if self._model._loaded_data_store.get_data(run=[run])]
             self._model._context.current_runs = run_list
 
-        self.update_view_from_model(run_list)
-        self._view.notify_loading_finished()
+            if self._load_multiple_runs and self._multiple_file_mode == "Co-Add":
+                run_list_to_add = [run for run in self.run_list if self._model._loaded_data_store.get_data(run=[run])]
+                run_list = [[run for run in self.run_list if self._model._loaded_data_store.get_data(run=[run])]]
+                load_utils.combine_loaded_runs(self._model, run_list_to_add)
+                self._model._context.current_runs = run_list
+
+            self.update_view_from_model(run_list)
+            self._view.notify_loading_finished()
+        except IndexError as error:
+            self._view.warning_popup(error)
         self.enable_loading()
```

Comment:
Could we add a `finally:` in before the re-enabling step just to ensure that if in the future we run into some unexpected exception we still re-enable the GUI afterwards?